lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -816,6 +915,14 @@ def main():
             if interface['type'] == 1:
                 ip = interface['ip']
 
+    if macros:
+        # convert macros to zabbix native format - {$MACRO}
+        for macro in macros:
+            if not macro['macro'].startswith('{$'):
+                macro['macro'] = '{$' + macro['macro']
+            if not macro['macro'].endswith('}'):
+                macro['macro'] = macro['macro'] + '}'
+
     # Use proxy specified, or set to 0
```

Comment:
You have to remove the None values from the list somehow:

```python3
if tags:
        # remove None values for api
        for tag in tags:
            if 'value' in tag and tag['value'] == None:
                tag.pop('value', True)
```

Reply:
That's why it is only Work In Progress! Thanks so much for looking into it. Gonna fix it asap so we can have it merged.

Reply:
Seems that zabbix defaults to empty strings so we need that as well in order to have idempotent results. Fix comming in a minute